liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerXmlDocument.java

```diff
@@ -176,32 +225,49 @@ public boolean accept(File file) {
         // Get matching files.
         File[] xmlFiles = dir.listFiles(filter);
         for (int i = 0; i < xmlFiles.length; i++) {
-            if (isFoundConfigInFragmentFile(xmlFiles[i].getPath())) {
+            if (isFoundAppConfigInFragmentFile(xmlFiles[i].getPath(), fileName)) {
                return true;
             }
         }
         return false;
     }
     
-    private static boolean isFoundConfigInFragmentFile(String filePath) {
+    private static boolean isFoundAppConfigInFragmentFile(String filePath, String fileName) {
         try {
-            if (isFoundTagName(filePath, "webApplication")) 
+            if (isFoundWebAppWithFileName(filePath, fileName)) {
                 return true;
+            }
         
-            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
-            DocumentBuilder builder = factory.newDocumentBuilder();
-            Document doc = builder.parse(filePath);
-            NodeList appList = doc.getElementsByTagName("application");
+            NodeList appList = getNodeList(filePath, "/server/application");
         
-            if (isFoundWebAppElement(appList))
+            if (isFoundAppConfigFromAppListByFileName(appList, fileName)) {
```

Comment:
It looks cleaner if you also have a `isFoundAppWithFileName(filePath, fileName)` method. see line 237